Interface/Harp.SoundCard/UpdateSoundWaveform.cs

```diff
@@ -0,0 +1,113 @@
+        public int? DeviceIndex { get; set; }
+
+        /// <summary>
+        /// Gets or sets the index of the sound to update.
+        /// </summary>
+        [Range(2, 31)]
+        [Editor(DesignTypes.NumericUpDownEditor, DesignTypes.UITypeEditor)]
+        [Description("The index of the sound to update.")]
+        public int SoundIndex { get; set; } = 2;
+
+        /// <summary>
+        /// Gets or sets the name of the sound to be stored in the device.
+        /// This field is optional.
+        /// </summary>
+        [Description("The name of the sound to be stored in the device. This field is optional.")]
+        public string SoundName { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value specifying the sample rate used to playback the
+        /// sound waveform.
+        /// </summary>
+        [Description("Specifies the sample rate used to playback the sound waveform.")]
+        public SampleRate SampleRate { get; set; }
+
+        /// <summary>
+        /// Replaces the specified sound waveform in the SoundCard device with each
+        /// of the sample buffers in an observable sequence.
+        /// </summary>
+        /// <param name="source">
+        /// A sequence of binary array objects representing all the raw samples of
+        /// the sound waveform. Continuous streaming is not supported.
+        /// </param>
+        /// <returns>
+        /// An observable sequence that is identical to the <paramref name="source"/> sequence
+        /// but where there is an additional side effect of replacing the sound waveform
+        /// of the specified sound with each of the sample buffers in the sequence.
+        /// </returns>
+        public override IObservable<byte[]> Process(IObservable<byte[]> source)
+        {
+            return source.Do(value =>
+            {
+                UpdateWaveform(DeviceIndex, SoundIndex, SampleRate, SampleType.Int32, value, SoundName);
+            });
+        }
+
+        /// <summary>
+        /// Replaces the specified sound waveform in the SoundCard device with each
+        /// of the sample buffers in an observable sequence.
+        /// </summary>
+        /// <param name="source">
+        /// A sequence of <see cref="Mat"/> objects representing all the raw samples of
+        /// the sound waveform. Continuous streaming is not supported.
+        /// </param>
+        /// <returns>
+        /// An observable sequence that is identical to the <paramref name="source"/> sequence
+        /// but where there is an additional side effect of replacing the sound waveform
+        /// of the specified sound with each of the sample buffers in the sequence.
+        /// </returns>
+        public IObservable<Mat> Process(IObservable<Mat> source)
+        {
+            return source.Do(value =>
+            {
+                if (value.Cols != 1 && value.Rows != 1 || value.Channels != 1)
```

Comment:
Not sure I follow this, can't the board take a stereo waveform? I thought it was possible... but maybe i am mistaken?

Reply:
This is a good point, I agree it definitely supports stereo. @filcarv @artursilva0 can you comment on this? Does the raw waveform always have to be stereo, or can it be mono as well? I couldn't see a flag anywhere in the CLI or API for this.